docs/libcurl/opts/CURLOPT_WRITEFUNCTION.3

```diff
@@ -109,6 +109,9 @@ For all protocols
 
  /* we pass our 'chunk' struct to the callback function */
  curl_easy_setopt(curl_handle, CURLOPT_WRITEDATA, (void *)&chunk);
+
+ /* remember to free the buffer */
+ free(chunk.response)
```

Comment:
Well, at this point in the code this `free()` call will do no good since it hasn't had the chance to allocate anything yet!

Shouldn't you then also perhaps invoke curl_easy_perform() before the free is called?

Reply:
Yes. I also fixed all remaining errors, i.e. `curl_handle` coming out of nowhere.